// obs-websocket-js/dist/msgpack.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); var _class;





var _chunkRL7MGYX4cjs = require('./chunk-RL7MGYX4.cjs');

// src/msgpack.ts
var _msgpack = require('@msgpack/msgpack');
var OBSWebSocket = (_class = class extends _chunkRL7MGYX4cjs.BaseOBSWebSocket {constructor(...args) { super(...args); _class.prototype.__init.call(this); }
  __init() {this.protocol = "obswebsocket.msgpack"}
  async encodeMessage(data) {
    return _msgpack.encode.call(void 0, data);
  }
  async decodeMessage(data) {
    if (typeof Blob !== "undefined" && data instanceof Blob) {
      data = await data.arrayBuffer();
    }
    return _msgpack.decode.call(void 0, data);
  }
}, _class);
var msgpack_default = OBSWebSocket;







exports.EventSubscription = _chunkRL7MGYX4cjs.EventSubscription; exports.OBSWebSocket = OBSWebSocket; exports.OBSWebSocketError = _chunkRL7MGYX4cjs.OBSWebSocketError; exports.RequestBatchExecutionType = _chunkRL7MGYX4cjs.RequestBatchExecutionType; exports.WebSocketOpCode = _chunkRL7MGYX4cjs.WebSocketOpCode; exports.default = msgpack_default;
